### DISCUSSION TITLE
Stuttering on large Images still present

### DISCUSSION THREAD

**Discussion by ghost** [mienaiyami/yomikiru#29] [General]
Until v2.3.5 I could see the stutter, but now the Stuttering has been fixed.

- Even while scrolling up or down very fast, there is no stutter now.
- Now I do not see any stutter except in large images of size (2000 x width).
- If you want to test it yourself, I can send the images to you.

By the way, what does scroll speed 1 and 2 do?

### Thank you for fixing the stuttering issue.

**Comment by mienaiyami** [comment]
Thank you for being patient for stuttering issue.
> If you want to test it yourself, I can send the images to you.

I tested with 4k images myself, so no need for that.
>By the way, what does scroll speed 1 and 2 do?

scroll 1 is w/s or arrow up/down (by default)
scroll 2 is space/ shift space (by default)

Now you can have 2 different scrolling speed on keypress.
Close issue if nothing to comment.

**Comment by ghost** [comment]
Please take a look at the below videos for reference on stutter that is happening to me on large images.

- Until v2.3.5, the same stutter happened for small images too.
- Now this only happens for large images.
- Image Type: webp
- Each Image Dimension: 2400 x width varies

I have sent you an invite to access the repository where I uploaded two videos of stutter, as they are more than 10MB I could not upload them here.

- Save them to your computer to play the videos.

https://github.com/alenwai/Stutter-Example

**Comment by mienaiyami** [comment]
Thats weird, because there is no stuttering for me now. Can you tell me your system spec, it might be the reason for stuttering. Or upload the folder whose video these are on https://github.com/alenwai/Stutter-Example, so I can test them on my system.

**Comment by ghost** [comment]
I uploaded the folder.

**Comment by mienaiyami** [comment]
Okay, so its stuttering for me as well. I will look into it. Sorry for your inconvenience

**Comment by ghost** [comment]
I tested many applications and found all of them to have the same stutter problem even in small images.

- I found one application to be an exception, [html mangareader](https://github.com/luejerry/html-mangareader).
- It is also written in typescript.  One of it's languages is typescript.
- Since it is written in the same language, you could refer it for a solution.

**Comment by mienaiyami** [comment]
Thanks for the suggestion, I will check it out.

**Comment by ghost** [comment]
Found a bunch of stuff online. If it is relevant to the problem.
1. [Chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=979890)
2. [Stack Overflow](https://stackoverflow.com/questions/56801191/how-to-convert-png-base-64-to-pixel-array-with-out-using-image-src-or-image-onlo)
3. [Stack Overflow](https://stackoverflow.com/questions/61345388/domexception-the-source-image-could-not-be-decoded)
4. [Smart Slider](https://smartslider.helpscoutdocs.com/article/2041-too-many-or-too-big-images-are-used)
5. [Mozilla](https://developer.mozilla.org/en-US/docs/Web/API/DOMException)
6. [Medium](https://medium.com/dailyjs/image-loading-with-image-decode-b03652e7d2d2)

**Comment by mienaiyami** [comment]
Thank you so much for providing these resources, I will try my best to fix it. Hopefully it is in this framework

**Comment by mienaiyami** [comment]
I'm very sorry, there might not be a solution for this, but I will keep trying.
As of now I recommend not using images with very high resolution (except its height) as you won't get greater quality unless your display is 4k and it will take lot of CPU usage to render and scale down to fit screen.
Sorry for not able to fix it. I'm only solo developers and I don't have any contacts with other developers.

**Comment by ghost** [comment]
> As of now I recommend not using images with very high resolution (except its height)

So the problem is not image height, but its width. I thought it was because of height.

> there might not be a solution for this, but I will keep trying.

Thank you.

> Sorry for not able to fix it. I'm only solo developers and I don't have any contacts with other developers.

- You already built an amazing application that takes low ram, has good clutter free UI, and performs way better than the popular applications.
- In v2.3.5 the scrolling is more smoother than the popular applications. I compared them side by side.
- You also said you will keep trying to find a solution for it. Maybe it will take time, but you could find the solution.

**Comment by mienaiyami** [comment]
I will turn it into discussion as i already have same issue registered.

**Comment by omodia** [comment]
> I found one application to be an exception, [html mangareader](https://github.com/luejerry/html-mangareader).

- After further testing, it is not true.
- When using chrome to read, the stuttering is more common. Example 100%.
- When using firefox to read, the stuttering is still present, but a little lesser impact than on chrome. Example 80%.
